File: Data Structures & Algorithms/reorder-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public void reorderList(ListNode head) {
        Stack<Integer> stack=new Stack<>();
        ListNode it=head;
        //Fill the stack
        while(it!=null){
            stack.push(it.val);
            it=it.next;
        }
        //Inject new nodes
        it=head;
        int n=stack.size();
        for(int i=0;i<n/2;i++){
            ListNode next=it.next;
            ListNode node= new ListNode(stack.pop(),next);
            it.next=node;
            it=next;
        }
        //Clean the Linked List
        it=head;
        int cur=0;
        while(it!=null){
            if(cur>=n-1){
                ListNode next=it.next;
                it.next=null;
                it=next;
            }else it=it.next;
            cur++;
        }
    }

}

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

public class Solution {
  public bool HasCycle(ListNode head)
  {
      ListNode current = head; 
      int length = 0;
      int index = 0; 
      List <ListNode> array = new List<ListNode>();
      while (current != null)
      {

          if( array.IndexOf(current) !=-1)
          {
              return true; 
          }
          array.Add(current);
          current = current.next;
      }


      return false;
  }
}
